draft-irtf-cfrg-hpke.md

```diff
@@ -273,50 +275,90 @@ values are two bytes long.
 
 ## DH-Based KEM
 
-Suppose we are given a Diffie-Hellman group that provides the
-following operations:
+Suppose we are given an instantiation KDF_kem of KDF with functions
+Hash_kem, Extract_kem, and Expand_kem, and a Diffie-Hellman group
+providing the following operations:
 
 - GenerateKeyPair(): Generate an ephemeral key pair `(sk, pk)`
   for the DH group in use
 - DH(sk, pk): Perform a non-interactive DH exchange using the
-  private key sk and public key pk to produce a shared secret
-  of length Npk
-- Marshal(pk): Produce a fixed-length byte string
+  private key sk and public key pk to produce a Diffie-Hellman
+  shared secret of length Ndh
+- Marshal(pk): Produce a byte string of length Npk
   encoding the public key `pk`
-- Unmarshal(enc): Parse a fixed-length byte string to recover a
+- Unmarshal(enc): Parse a byte string of length Npk to recover a
   public key
+- Nzz: The output size of the Hash_kem and Extract_kem functions
```

Comment:
Re-defining this is somewhat confusing -- is it needed?

Reply:
We also redefine GenerateKeyPair, DH, Marshal, Unmarshal to make their definitions adapted to DHKEM. With the line about Nzz I want to make clear that Nzz is the length of Hash_kem's output. This is only implicitly clear by seeing that the result of Expand is returned.

It's true that this (Nzz is the length of Hash_kem's output) is true for our variants of DHKEM with SHA2. But now that I included Nzz as third argument to Expand, other settings are also possible.

So maybe we could move this line away from the definitions list and put it as a comment below the pseudo-code, saying that for our specified versions of DHKEM, Nzz is Hash_kem's output length. What do you think about that?

Reply:
> So maybe we could move this line away from the definitions list and put it as a comment below the pseudo-code, saying that for our specified versions of DHKEM, Nzz is Hash_kem's output length. What do you think about that?

That'd be great!